### ISSUE TITLE
[stmt.if] p1 The "shall" is a bit misleading 

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/CWG#270]
Full name of submitter (unless configured in github; will be published with the issue): Jim X

[stmt.if] p1 says
> In the second form of if statement (the one including else), if the first substatement is also an if statement then that inner if statement shall contain an else part.

Assume this case is the mentioned one:
````cpp
if(false) if(true);else{}
````
The whole if statement is always analyzed to the first form. The corresponding note implies this point
> 77) In other words, the else is associated with the nearest un-elsed if.

The example is well-formed. However, "shall" in the wording sounds like if we don't give `if(true);` the additional `else` part, it will violate the rule. Furthermore, as aforementioned, the if statement is never subsumed to the second form, so the wording is a bit misleading.

**Comment by eisenwave** [comment]
I think this is editorial. The mechanics of what happens are pretty clear, but nowadays, we probably wouldn't say "shall" here but that the `else` part is attached to the innermost `if`.

The wording does seem a bit clunky, but given that there's no real confusion/implementation divergence here, I don't see any urgent need to touch it.

**Comment by jensmaurer** [comment]
I agree the "shall" is sub-optimal, but it seems a sightly more aggressive fix than turning "shall" into "is" should be applied, maybe explicitly highlighting the parsing ambiguity and the resolution of such.  Specific suggestions welcome.

**Comment by frederick-vs-ja** [comment]
CWG3212 also addresses this.
